src/services/programService/httpClient.js

```diff
@@ -27,14 +27,116 @@ import fetch from 'node-fetch';
 import { PROGRAM_SERVICE_HTTP_ROOT } from '../../config';
 import { restErrorResponseHandler } from '../../utils/restUtils';
 
-const getProgramPublicFields = async (programShortName) => {
-	const url = `${PROGRAM_SERVICE_HTTP_ROOT}/public/program?name=${programShortName}`;
-	const response = await fetch(url, {
+//data formatters
+const formatPublicProgram = (program) => ({
+	name: program.name,
+	shortName: program.shortName,
+	description: program.description,
+	website: program.website,
+	institutions: program.programInstitutions?.map((institution) => institution.name) || [],
+	countries: program.programCountries?.map((country) => country.name) || [],
+	regions: program.processingRegions?.map((region) => region.name) || [],
+	cancerTypes: program.programCancers?.map((cancer) => cancer.name) || [],
+	primarySites: program.programPrimarySites?.map((primarySite) => primarySite.name) || [],
+});
+
+const formatPrivateProgram = (program) => {
+	return program.program;
+};
+
+const formatPrivateProgramList = (programList) => programList.map(formatPrivateProgram);
+
+const formatJoinProgramInvite = (invitation) => {
+	const formattedObj = {
+		...invitation,
+		createdAt: new Date(invitation.createdAt),
+		expiresAt: new Date(invitation.expiresAt),
+		acceptedAt: new Date(invitation.acceptedAt),
+		user: { ...invitation.user, role: invitation.user.role.value },
+		program: {
+			...invitation.program,
+			institutions: invitation.program.programInstitutions,
+			countries: invitation.program.programCountries,
+			cancerTypes: invitation.program.programCancers,
+			primarySite: invitation.program.programPrimarySites,
+		},
+	};
+
+	delete formattedObj.program.programInstitutions;
+	delete formattedObj.program.programCountries;
+	delete formattedObj.program.programCancers;
+	delete formattedObj.program.programPrimarySites;
+	return formattedObj;
+};
+
+//private fields
+export const listPrivatePrograms = async (jwt = null) => {
+	const url = `${PROGRAM_SERVICE_HTTP_ROOT}/programs`;
```

Comment:
I would recommend using `url-join` to build URLs, like here: https://github.com/icgc-argo/platform-api/blob/develop/src/services/clinical/api.ts#L42

This can fix some bugs that occur when the service domain is incorrectly formatted (sometimes has a trailing slash, sometimes doesnt... with urlJoin we don't care which).

Reply:
urljoin is implemented. Thanks for showing me this tool! I read the benefits to prevent errors using it!